private/buf/cmd/buf/command/mod/modupdate/modupdate.go

```diff
@@ -165,6 +166,10 @@ func run(
 	}
 
 	if err := bufmoduleref.PutDependencyModulePinsToBucket(ctx, readWriteBucket, dependencyModulePins); err != nil {
+		var inconsistentDigestErr bufmoduleref.InconsistentDigestError
+		if errors.As(err, &inconsistentDigestErr) {
+			return err
```

Comment:
The mechanics of this can change, but what I'm trying to avoid here is printing out an error message that indicates it is an internal error that needs to be reported to Buf's issues page for a well known error.